### DISCUSSION TITLE
Why do the wrapper objects use abstract classes instead of interfaces?

### DISCUSSION THREAD

**Discussion by tacosontitan** [LuckyPennySoftware/MediatR#803] [Q&A] [answered]
Currently, all three of [the wrapper files](https://github.com/jbogard/MediatR/tree/master/src/MediatR/Wrappers) contain abstract definitions and a corresponding implementation such as:
```csharp
public abstract class SampleWrapper {
    public abstract void HelloWorld();
}
public class SampleWrapperImpl : SampleWrapper {
    public override void HelloWorld() =>
        Console.WriteLine("Hello world!");
}
```
This could have been an interface and default implementation instead such as:
```csharp
public interface ISampleWrapper {
    void HelloWorld();
}
public class SampleWrapper : ISampleWrapper {
    public void HelloWorld() =>
        Console.WriteLine("Hello world!");
}
```
What was the driving factor to use an abstract class over an interface when no partial implementations were provided?

**Comment by jbogard** [comment, marked as answer]
Those wrapper classes are there to [bridge the generics divide](https://jimmybogard.com/crossing-the-generics-divide/). That's all.
